lisp/mastodon-http.el

```diff
@@ -82,21 +83,32 @@ Authorization header is included by default unless UNAUTHENTICED-P is non-nil."
                       "&")))
         (url-request-extra-headers
 	 (append
-	  (unless unauthenticed-p
+	  (unless (or unauthenticed-p
+                      (eq 'oauth2 mastodon-auth-mechanism))
```

Comment:
That indentation looks weird.

When you are done, can you run `M-x untabbify` on all touched files to get rid of these pesky tab characters?